src/main/kotlin/rx/lang/kotlin/observables.kt

```diff
@@ -1,107 +1,98 @@
 package rx.lang.kotlin
 
-import rx.Observable
-import rx.Subscriber
-import rx.Subscription
-import rx.observables.BlockingObservable
+import io.reactivex.Observable
+import io.reactivex.ObservableEmitter
+import io.reactivex.Single
+import io.reactivex.disposables.Disposable
+import io.reactivex.functions.BiFunction
 
-fun <T> emptyObservable() : Observable<T> = Observable.empty()
-fun <T> observable(body : (s : Subscriber<in T>) -> Unit) : Observable<T> = Observable.create(body)
 /**
- * Create deferred observable
- * @see [rx.Observable.defer] and [http://reactivex.io/documentation/operators/defer.html]
+ * [Observable.empty] alias
  */
-fun <T> deferredObservable(body : () -> Observable<T>) : Observable<T> = Observable.defer(body)
-private fun <T> Iterator<T>.toIterable() = object : Iterable<T> {
-    override fun iterator(): Iterator<T> = this@toIterable
-}
+fun <T : Any> emptyObservable(): Observable<T> = Observable.empty()
 
-fun BooleanArray.toObservable() : Observable<Boolean> = this.toList().toObservable()
-fun ByteArray.toObservable() : Observable<Byte> = this.toList().toObservable()
-fun ShortArray.toObservable() : Observable<Short> = this.toList().toObservable()
-fun IntArray.toObservable() : Observable<Int> = this.toList().toObservable()
-fun LongArray.toObservable() : Observable<Long> = this.toList().toObservable()
-fun FloatArray.toObservable() : Observable<Float> = this.toList().toObservable()
-fun DoubleArray.toObservable() : Observable<Double> = this.toList().toObservable()
-fun <T> Array<out T>.toObservable() : Observable<T> = Observable.from(this)
-
-fun IntProgression.toObservable() : Observable<Int> =
-        if (step == 1 && last.toLong() - first < Integer.MAX_VALUE) Observable.range(first, Math.max(0, last - first + 1))
-        else Observable.from(this)
+/**
+ * [Observable.defer] alias
+ */
+fun <T : Any> Observable<T>.defer() = Observable.defer { this }
```

Comment:
This shouldn't exist. The whole point of defer is the function is deferred to create an Observable. If you already have one there's no point in calling defer.